// Copyright 2023 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";


enum IamAlphaWorkforcePoolProviderStateEnum {
  IamAlphaWorkforcePoolProviderStateEnumNO_VALUE_DO_NOT_USE = 0;
  IamAlphaWorkforcePoolProviderStateEnumSTATE_UNSPECIFIED = 1;
  IamAlphaWorkforcePoolProviderStateEnumACTIVE = 2;
  IamAlphaWorkforcePoolProviderStateEnumDELETED = 3;
}

message IamAlphaWorkforcePoolProvider {
  string name = 1;
  string display_name = 2;
  string description = 3;
  IamAlphaWorkforcePoolProviderStateEnum state = 4;
  bool disabled = 5;
  map<string, string> attribute_mapping = 6;
  string attribute_condition = 7;
  IamAlphaWorkforcePoolProviderSaml saml = 8;
  IamAlphaWorkforcePoolProviderOidc oidc = 9;
  string location = 10;
  string workforce_pool = 11;
}

message IamAlphaWorkforcePoolProviderSaml {
  string idp_metadata_xml = 1;
}

message IamAlphaWorkforcePoolProviderOidc {
  string issuer_uri = 1;
  string client_id = 2;
}

message ApplyIamAlphaWorkforcePoolProviderRequest {
  IamAlphaWorkforcePoolProvider resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteIamAlphaWorkforcePoolProviderRequest {
  string service_account_file = 1;
  IamAlphaWorkforcePoolProvider resource = 2;
}

message ListIamAlphaWorkforcePoolProviderRequest {
  string service_account_file = 1;
  string Location = 2;
  string WorkforcePool = 3;
}

message ListIamAlphaWorkforcePoolProviderResponse {
  repeated IamAlphaWorkforcePoolProvider items = 1;
}

service IamAlphaWorkforcePoolProviderService {
  rpc ApplyIamAlphaWorkforcePoolProvider(ApplyIamAlphaWorkforcePoolProviderRequest) returns (IamAlphaWorkforcePoolProvider);
  rpc DeleteIamAlphaWorkforcePoolProvider(DeleteIamAlphaWorkforcePoolProviderRequest) returns (google.protobuf.Empty);
  rpc ListIamAlphaWorkforcePoolProvider(ListIamAlphaWorkforcePoolProviderRequest) returns (ListIamAlphaWorkforcePoolProviderResponse);
}
